semgrep-core/src/typing/LSP_client.ml

```diff
@@ -21,26 +21,27 @@ open Types
 module PI = Parse_info
 module G = AST_generic
 
+let logger = Logging.get_logger [ __MODULE__ ]
+
 (*****************************************************************************)
 (* Prelude *)
 (*****************************************************************************)
 
 (*****************************************************************************)
 (* Types and globals *)
 (*****************************************************************************)
-type env = { io : Io.t option; last_uri : string }
-
-let global = ref { io = None; last_uri = "" }
+type env = { io : Io.t option; last_uri : Uri.t }
 
-let debug = ref false
+let global = ref { io = None; last_uri = Uri.of_path "" }
 
 (*****************************************************************************)
 (* Helpers *)
 (*****************************************************************************)
 
 let server =
-  match 1 with
-  | 1 -> "/home/pad/.opam/4.09.1/bin/ocamllsp"
+  match 0 with
+  | 0 -> "ocamllsp"
+  | 1 -> "/home/pad/.opam/4.12.0/bin/ocamllsp"
```

Comment:
We should use `Sys.getenv "HOME"` here.

Reply:
Ideally we should search for the binary in the `$PATH`.

Reply:
yes, this is hardcoded for now because it's very experimental.